sumfields.php

```diff
@@ -290,63 +290,68 @@ function sumfields_civicrm_triggerInfo(&$info, $tableName) {
   $sql_field_parts = array();
 
   $active_fields = sumfields_get_setting('active_fields', array());
+  $process_type = sumfields_get_setting('data_update_method','default');
 
   $session = CRM_Core_Session::singleton();
 
-  // Iterate over all our fields, and build out a sql parts array
-  while(list($base_column_name, $params) = each($custom_fields)) {
-    if(!in_array($base_column_name, $active_fields)) continue;
-    $table = $custom['fields'][$base_column_name]['trigger_table'];
-    if(!is_null($tableName) && $tableName != $table) {
-      // if triggerInfo is called with particular table name, we should
-      // only respond if we are contributing triggers to that table.
-      continue;
+  if ($process_type == 'via_triggers') {
+
```

Comment:
And again $data_update_method

Reply:
And - could we move this check to the top of the function and check if != 'via_triggers' and return. we don't need to execute any of that code at the top of the function if we not using the via_triggers approach.